File: moviejava/app/src/main/java/com/example/moviejava/data/remote/people/PeopleRepository.java
package com.example.moviejava.data.remote.people;


import com.example.moviejava.model.JsonObject;
import com.example.moviejava.model.Person;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Observable;
import retrofit2.Retrofit;

@Singleton
public class PeopleRepository {
    private PeopleService service;

    @Inject
    public PeopleRepository(Retrofit retrofit) {
        service = retrofit.create(PeopleService.class);
    }

    public @NonNull Observable<Person> getPerson(int id) {
        return service.getPerson(id);
    }

    public @NonNull Observable<JsonObject> getMovieByPerson(int id) {
        return service.getMovieByPerson(id);

    }
}

File: movie/app/src/main/java/com/example/movie/data/remote/people/PeopleRepository.kt
package com.example.movie.data.remote.people

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import retrofit2.Retrofit
import javax.inject.Inject

class PeopleRepository @Inject constructor(val retrofit: Retrofit) {
    private val service = retrofit.create(PeopleService::class.java)
    fun getPerson(id: Int) = flow {
        emit(service.getPerson(id))
    }.flowOn(Dispatchers.IO)

    fun getMovieByPerson(id: Int) = flow {
        emit(service.getMovieByPerson(id))
    }.flowOn(Dispatchers.IO)
}
